data_and_models/pipelines/ner/clean.py

```diff
@@ -0,0 +1,121 @@
+
+import srsly
+import typer
+
+
+def is_valid(example: Dict[str, Any], duplicated_hashes: set) -> bool:
+    accepted = example["answer"] == "accept"
+    unique_hash = example["_input_hash"] not in duplicated_hashes
+    return all((accepted, unique_hash))
+
+
+def main(
+    input_path: Path = typer.Argument(..., exists=True, dir_okay=False),
+    keep_label: str = typer.Argument(...),
+    rename_into: Optional[str] = typer.Argument(None),
+):
+    keep_label = keep_label.upper()
+    rename_into = None if rename_into is None else rename_into.upper()
+
+    print("Read annotations...")
+    corpus = list(srsly.read_jsonl(input_path))
+    print(f"...read {len(corpus)} texts")
+
+    print("Identify duplicated 'input_hash'...")
+    seen_hashes = set()
+    duplicated_hashes = set()
+    for x in corpus:
+        hash = x["_input_hash"]
+        if hash in seen_hashes:
+            duplicated_hashes.add(hash)
+        else:
+            seen_hashes.add(hash)
+    hashes_count = len(duplicated_hashes)
+    print(f"...identified {hashes_count} hash(es)")
+    if hashes_count > 0:
+        debug = " ".join(map(str, sorted(duplicated_hashes)))
+        print(f">>> {debug}")
+
+    print("Keep only valid texts...")
+    valid_texts = [x for x in corpus if is_valid(x, duplicated_hashes)]
+    print(f"...kept {len(valid_texts)} texts")
+    print(f"...that's {len(valid_texts) / len(corpus):.0%} of the total")
+
+    print("Normalize labels...")
+    total_spans = 0
+    normalized_spans = 0
+    for text in valid_texts:
+        for span in text.get("spans", []):
+            label = span["label"]
+            total_spans += 1
+            if label.islower():
+                span["label"] = label.upper()
+                normalized_spans += 1
+    print(f"...normalized {normalized_spans} label spans (on {total_spans})")
+    print(f"...that's {normalized_spans/total_spans:.0%} of the total")
+
+    renaming = "" if rename_into is None else f" (renamed into {rename_into})"
+    print(f"Keep only label {keep_label}{renaming}...")
+    output_texts = []
+    kept_spans = 0
+    for text in valid_texts:
+        filtered_spans = []
+        for span in text.get("spans", []):
+            if span["label"] == keep_label:
+                if rename_into is not None:
+                    span["label"] = rename_into
+                filtered_spans.append(span)
+        text["spans"] = filtered_spans
+        kept_spans += len(text["spans"])
+        output_texts.append(text)
+    print(f"...kept {kept_spans} label spans (on {total_spans})")
+    print(f"...that's {kept_spans / total_spans:.0%} of the total")
+
+    print("Write cleaned annotations...")
+    output_label = keep_label if rename_into is None else rename_into
+    output_path = input_path.parent / f"annotations_{output_label.lower()}.jsonl"
+    srsly.write_jsonl(output_path, output_texts)
+    print(f"...wrote {output_path}")
+
+
+if __name__ == "__main__":
+    typer.run(main)
```

Comment:
Is there a specific reason to user `typer` package here ? I just checked and all the other scripts (`data_and_models/pipelines/ner/eval.py`, `train.py`, ...) except the two introduced by this PR and `preprocess.py`are using:
```python
if __name__ == "__main__":
     main()
```
and `argparse` for the CLI arguments. 

Do you think this package is useful/necessary ? If yes, it could be nice to open a ticket to harmonize the other scripts in that sense in my opinion. 
Moreover, I see that `typer` is also in the `INSTALL_REQUIRES` of the `setup.py` and the main `requirements.txt`. Should it not be part of the `data-and-models` logic ?

Reply:
We could harmonize to `typer` or add complexity by using `argparse`.

I think it was not moved by #345 to extra. But it could.

Reply:
Hello @EmilieDel,

Regarding the leftovers from #345, I have created a dedicated issue to avoid out-of-scope work in this PR:
https://github.com/BlueBrain/Search/issues/368

Reply:
As discussed yesterday, I have moved the scripts `analyze.py` and `clean.py` from typer to argparse.

Done in 1ede058d51fa81ad3c26407cfaba3caaffa1a8e9.